### ISSUE TITLE
Set Node v20 as default version

### ISSUE THREAD

**Issue by ThimoDEV** [railwayapp/nixpacks#1144]
### Feature request

Node v18 is currently in maintenance mode, so for me it would make sense to have v20 as the default.

Is there a reason its by default still on v18?

### Motivation

For Lucia auth I have to use Node v20 and I know I can override it, but it would be a minor upgrade but convenient to have.

### Contribution

I could sent a PR if that helps, although I believe its a one line change?

**Comment by coffee-cup** [comment]
In most cases we recommend setting the Node version in a `package.json` or similar file. I think it would be a good idea to update the default soon though.

At Railway, we are holding off updating the default until an internal project is complete that prevents breaking existing apps that require the Node 18 default.

**Comment by EvHaus** [comment]
Setting this in `package.json`:

```json
"engines": {
  "node": ">=20.9.0"
}
```

Doesn't seem to have any effect on which version Railway uses to build the project. Seeing this in the build logs:

```
load build definition from railpack-plan.json
install apt packages: nodejs
...
Get:9 http://deb.debian.org/debian bookworm/main amd64 nodejs-doc all 18.20.4+dfsg-1~deb12u1 [3579 kB]
...
Unpacking nodejs (18.20.4+dfsg-1~deb12u1) ...
```

Is there a different way I could tell Railway to use a specific Node version (other than writing the whole Dockerfile myself)?

**Note**: Next.js 16 was released today, which requires Node 20.9.0+, so I expect more Next.js Railway users will start running into this in the next few days.

**Comment by ravi-sohal** [comment]
I have tried updating package.json engines.node to ">=20.9.0" as mentioned above but had no luck. Getting the same error - 

<img width="565" height="108" alt="Image" src="https://github.com/user-attachments/assets/8987e32e-7b7a-4621-b976-5338b77e7c4d" />

**Comment by iloveitaly** [comment]
@ravi-sohal 👋 I'm helping out with nixpacks/railpack maintenance!

We are deprecating nixpacks and replacing it with [railpack](https://github.com/railwayapp/railpack) (which supports ~all node versions). We actually [just landed support](https://github.com/railwayapp/railpack/commit/55e72e4e8b9f3a5023f9e520ceff1e77f4ff4d53) for pulling the version from the `engines` field (although a new version is not cut yet).

Could you give railpack a shot and let me know if you run into any issues?

**Comment by ravi-sohal** [comment]
@iloveitaly @EvHaus - I was able to complete my deployment.

Try any or all of the following items -

- Add an environment variable `NIXPACKS_NODE_VERSION="22"`
- Create an `.nvmrc` file in your root directory and add `20` to it
- Add the following value to package.json engines property `"node": ">=20.9.0 <21"`

**Comment by EvHaus** [comment]
Still not working for me when using `engines` property. Link to build here: https://railway.com/project/123a32e7-7fcb-4138-a5fe-6ed5bd6a528a/service/4cc205d8-6ffa-4699-96e2-9d888656ab02?environmentId=2040f731-2a2d-4374-8fd9-c27fd6407eb0&id=ea42d3b0-e29c-46bd-bbd4-d4d04c97e654#build

Railpack 0.9.2

package: bun  │  1.3.1 

Fails with:

```
$ next build
You are using Node.js 18.20.4. For Next.js, Node.js version ">=20.9.0" is required
```

Adding `NIXPACKS_NODE_VERSION` also seemed to have no effect. Could it be that the bun 1.3.1 nixpack package just has that old Node version inside it?

**Comment by ravi-sohal** [comment]
@EvHaus have you tried creating a .nvmrc file with `20` in it?

**Comment by iloveitaly** [comment]
@EvHaus the problem is the ENV vars have changes from nixpacks => railpack.

[Here's the documentation](https://railpack.com/languages/node) on how to configure your bun version. `RAILPACK_BUN_VERSION` is going to be the easiest way for you (if you are using bun). We've also improved version detection a bunch (and still have further improvements in flight), so you can use `.bun-version` and other version definitions as well.

**Comment by EvHaus** [comment]
I see Railpack is already using the right Bun version, so I don't think setting the Bun version is going to do anything here.

I set `RAILPACK_NODE_VERSION` to 22:

<img width="616" height="258" alt="Image" src="https://github.com/user-attachments/assets/4a39f733-9346-4c1a-870f-f10ca5457f59" />

And then triggered a rebuild, but got the same problem:

<img width="891" height="1246" alt="Image" src="https://github.com/user-attachments/assets/f9016b01-fca2-4525-9e1c-024eb67e874d" />

It seems like Railpack's Bun image has an old version of Node inside it or something?

Adding an `.nvmrc` to my repo **DID** help though. Running into new build problems now, but at least it seems like the Node version mismatch went away.

**Comment by nghiepdev** [comment]
I'm seeing the same issue with both Nixpack and Railpack on Dokploy 0.25.6. When deploying with Bun, the Node.js version is always 18, even after setting ‎⁠RAILPACK_NODE_VERSION=24⁠ and ‎⁠NIXPACKS_NODE_VERSION=24⁠.

**Comment by iloveitaly** [comment]
@EvHaus @nghiepdev Ah, I see the [issue on our end](https://github.com/railwayapp/railpack/blob/b5666f37246d68a314a3e540ac12a453ba592793/core/providers/node/node.go#L326-L328). To make sure I have this right:

* Bun is being used to run the server
* Node is being used as part of the build process (`next build`)
* Bun version is being picked up correctly
* `RAILPACK_NODE_VERSION` is set to the correct version
* Incorrect node version is being used

I don't have a quick fix for you right now, but will get this fixed next week.

**Comment by EvHaus** [comment]
@iloveitaly Yes -- I think that sounds about right although I'm not sure why Node is being used as part of the build process. Railpack is correctly detecting that I'm using `bun` and so I would have expected it to call `next build` using `bun`, rather than `node`. That could be another way of solving this but that might be too restrictive for our Railway users.

**Comment by RadheyMugdal** [comment]
using **nvmrc** solved the problem in my case. adding variables doesn't worked in my case.

**Comment by Psycarlo** [comment]
Using `engines.node`, `RAILPACK_NODE_VERSION` and `NIXPACKS_NODE_VERSION` did not work for me.
Only `.nvmrc`

**Comment by iloveitaly** [comment]
@Psycarlo Were you having trouble with the version of node installed alongside bun, or node without bun? Right now they use two different resolution paths, although that is very confusing and should be fixed.

**Comment by jayleaton** [comment]
Was there a fix for this? I am unable to deploy the latest version of my production apps because it is defaulting to node 18 after a next. 16 update?

EDIT:
Setting `RAILPACK_NODE_VERSION=21` as an env variable seems to work for npm or pnpm setups after updating to next.js 16
